universe/node/api_object.go

```diff
@@ -582,3 +583,122 @@ func (n *Node) apiObjectsGetObjectSubOptions(c *gin.Context) {
 
 	c.JSON(http.StatusOK, out)
 }
+
+// @Summary Claim and customise object
+// @Schemes
+// @Description Claim and customise object
+// @Tags objects
+// @Accept json
+// @Produce json
+// @Param object_id path string true "Object UMID"
+// @Param body body node.apiClaimAndCustomise.Body true "body params"
+// @Success 200 {object} nil
+// @Failure 500 {object} api.HTTPError
+// @Failure 400 {object} api.HTTPError
+// @Failure 403 {object} api.HTTPError
+// @Failure 404 {object} api.HTTPError
+// @Router /api/v4/objects/{object_id}/claim-and-customise [post]
+func (n *Node) apiClaimAndCustomise(c *gin.Context) {
+	type Body struct {
+		Title     string `json:"title" binding:"required"`
+		Text      string `json:"text" binding:"required"`
+		ImageHash string `json:"image_hash" binding:"required"`
+	}
+
+	var inBody Body
+	if err := c.ShouldBindJSON(&inBody); err != nil {
+		err = errors.WithMessage(err, "Node: apiClaimAndCustomise: failed to bind json")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_request_body", err, n.log)
+		return
+	}
+
+	objectID, err := umid.Parse(c.Param("objectID"))
+	if err != nil {
+		err := errors.WithMessage(err, "Node: apiClaimAndCustomise: failed to parse object umid")
+		api.AbortRequest(c, http.StatusBadRequest, "invalid_object_id", err, n.log)
+		return
+	}
+
+	object, ok := n.GetObjectFromAllObjects(objectID)
+	if !ok {
+		err := errors.Errorf("Node: apiClaimAndCustomise: object not found: %s", objectID)
+		api.AbortRequest(c, http.StatusNotFound, "object_not_found", err, n.log)
+		return
+	}
+
+	userID, err := api.GetUserIDFromContext(c)
+	if err != nil {
+		err = errors.WithMessage(err, "Node: apiClaimAndCustomise: failed to get user umid")
+		api.AbortRequest(c, http.StatusInternalServerError, "get_user_id_failed", err, n.log)
+		return
+	}
+
+	if object.GetOwnerID() == userID {
```

Comment:
yeah, don't think we need this limitation. Why can't I just 'claim' it as the creator to show as an example.

Reply:
removed